src/gc/gc.cpp

```diff
@@ -736,7 +737,7 @@ class t_join
 #endif //JOIN_STATS
 
         assert (!join_struct.joined_p);
-        int color = join_struct.lock_color;
+        int color = join_struct.lock_color.LoadWithoutBarrier();
```

Comment:
@ChrisAhna @maoni This is relying on the lock_color Volatile store release barrier when it is written to flush out its write.  .  This will ultimately cause the write to invalidate the other processors cache lines.  It is relying to on the Volatile<T>.LoadWithoutBarrier() to guarantee the load is treated as volatile and is not moved out of the loop.